File: LeetCode/383 Ransom Note.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) {
        int[] count = new int[26];
        for (char c: magazine.toCharArray()) {
            count[c - 'a'] += 1;
        }
        for (char c: ransomNote.toCharArray()) {
            if (count[c - 'a'] == 0) { return false; }
            count[c - 'a'] -= 1;
        }
        return true;
    }
}

File: LeetCode/383 Ransom Note.py

class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        magazine_counter = Counter(magazine)
        ransom_counter = Counter(ransomNote)
        
        for c in ransom_counter:
            if c not in magazine_counter or magazine_counter[c]<ransom_counter[c]:
                return False
        return True